compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp

```diff
@@ -70,10 +69,182 @@ void compute_asym_scale_zp(float min, float max, float &scaling_factor, int64_t
   zp = nudged_zero_point;
 }
 
+void get_channel_dim_index(CircleConst *node, int *len_per_dim, int &channel_dim_index)
+{
+  auto succs = loco::succs(node);
+  if (succs.size() != 1) // assume weights is used by only one node
+    return;
+
+  for (auto out : succs)
+  {
+    auto conv = dynamic_cast<CircleConv2D *>(out);
+    auto dw_conv = dynamic_cast<CircleDepthwiseConv2D *>(out);
+    auto tw_conv = dynamic_cast<CircleTransposeConv *>(out);
+    auto fc = dynamic_cast<CircleFullyConnected *>(out);
+
+    if (conv != nullptr && conv->filter() == node)
+    {
+      len_per_dim[0] = node->dim(0).value();
+      len_per_dim[1] = node->dim(1).value();
+      len_per_dim[2] = node->dim(2).value();
+      len_per_dim[3] = node->dim(3).value();
+      channel_dim_index = 3;
+    }
+    else if ((tw_conv != nullptr && tw_conv->filter() == node) ||
+             (dw_conv != nullptr && dw_conv->filter() == node))
+    {
+      len_per_dim[0] = node->dim(0).value();
+      len_per_dim[1] = node->dim(1).value();
+      len_per_dim[2] = node->dim(2).value();
+      len_per_dim[3] = node->dim(3).value();
+      channel_dim_index = 2;
+    }
+    else if (fc != nullptr && fc->weights() == node)
+    {
+      len_per_dim[0] = 1;
+      len_per_dim[1] = 1;
+      len_per_dim[2] = node->dim(0).value();
+      len_per_dim[3] = node->dim(1).value();
+      channel_dim_index = 3;
+    }
```

Comment:
Please add an assertion to ensure that this function is called for the Ops supporting channel-wise quant.
E.g.,
```
else
{
  assert(false);
}
```